strax/processing/hitlets.py

```diff
@@ -388,31 +391,36 @@ def get_fwxm(hitlet, fraction=0.5):
         right last sample + 1.
     """
     data = hitlet['data'][:hitlet['length']]
-    max_val = hitlet['amplitude'] * fraction
 
     index_maximum = np.argmax(data)
-    pre_max = data[:index_maximum]
-    post_max = data[1 + index_maximum:]
-
-    # First the left edge:
-    lbi, lbs = _get_fwxm_boundary(pre_max, max_val)  # coming from the left
-    if lbi == NO_FWXM:
-        # We have not found any sample below:
-        left_edge = 0.
-    else:
-        # We found a sample below so lets compute
-        # the left edge:
+    max_val = data[index_maximum] * fraction
+    if np.all(data > max_val) or np.all(data == 0):
+        # In case all samples are larger than there is now FWXM by definition.
+        return np.nan, np.nan
+
+    pre_max = data[:index_maximum]  # Does not include maximum
+    post_max = data[1 + index_maximum:]  # same
+
+    if len(pre_max) and np.any(pre_max <= max_val):
+        # First the left edge:
+
+        lbi, lbs = _get_fwxm_boundary(pre_max[::-1], max_val)  # Reversing data starting at sample
+        # before maximum and go left
+        lbi = (index_maximum - 1) - lbi  # start sample minus samples we went to the left
         m = data[lbi + 1] - lbs  # divided by 1 sample
-        left_edge = lbi + (max_val - lbs) / m + 0.5  # .5 to start from bin center
+        left_edge = lbi + (max_val - lbs) / m + 0.5
```

Comment:
I don't see you addressed https://github.com/AxFoundation/strax/issues/348 or maybe I missed it. Can you also point me to the test that should be able to catch it in the future?

Reply:
Okay sorry, valid point I should have elaborated this a little bit more. I changed a bit the way how the the values for `lbi`and `lbs` are found. In the previous version it was possible  that  `data[lbi + 1] == lbs` are the same . This is now excluded, partially due to a change on how these values are found and partially due to some additional constraints to the data in case the FWHM is not defined.

Reply:
Thanks, are they tested against explicitly?

Reply:
Now it is tested explicitly.